benchmarks/bench.cpp

```diff
@@ -1,18 +1,74 @@
 #include <iostream>
 #include <boost/url/src.hpp>
+#include <uriparser/Uri.h>
+
+#include "ada.h"
+
 #include <benchmark/benchmark.h>
 using namespace boost::urls;
 
+
+/**
+ * Realistic URL examples collected on the actual web.
+ */
+std::string url_examples[] = {
+    "https://www.google.com/"
+    "webhp?hl=en&amp;ictx=2&amp;sa=X&amp;ved=0ahUKEwil_"
+    "oSxzJj8AhVtEFkFHTHnCGQQPQgI",
+    "https://support.google.com/websearch/"
+    "?p=ws_results_help&amp;hl=en-CA&amp;fg=1",
+    "https://en.wikipedia.org/wiki/Dog#Roles_with_humans",
+    "https://www.tiktok.com/@aguyandagolden/video/7133277734310038830",
+    "https://business.twitter.com/en/help/troubleshooting/"
+    "how-twitter-ads-work.html?ref=web-twc-ao-gbl-adsinfo&utm_source=twc&utm_"
+    "medium=web&utm_campaign=ao&utm_content=adsinfo",
+    "https://images-na.ssl-images-amazon.com/images/I/"
+    "41Gc3C8UysL.css?AUIClients/AmazonGatewayAuiAssets",
+    "https://www.reddit.com/?after=t3_zvz1ze",
+    "https://www.reddit.com/login/?dest=https%3A%2F%2Fwww.reddit.com%2F"};
```

Comment:
Can you also include ipv4 and ipv6?

Reply:
I suggest we produce *several* benchmarks. This one, called 'basic', should represent 'common/realistic' cases.